packages/gamut-labs/src/AppHeader/AppHeaderElements/types.tsx

```diff
@@ -2,78 +2,76 @@ import { GamutIconProps } from '@codecademy/gamut-icons';
 import { ReactNode } from 'react';
 
 export type AppHeaderItem =
-  | AppHeaderLogoItem
-  | AppHeaderLinkItem
-  | AppHeaderTextButtonItem
-  | AppHeaderFillButtonItem
+  | AppHeaderItemWithHref
   | AppHeaderSimpleDropdownItem
   | AppHeaderProfileDropdownItem
   | AppHeaderRenderElementItem;
 
-type AppHeaderBaseItem = {
+type AppHeaderItemWithHref =
+  | AppHeaderLogoItem
+  | AppHeaderLinkItem
+  | AppHeaderTextButtonItem
+  | AppHeaderFillButtonItem;
+
+type AppHeaderBaseItem<T extends string> = {
```

Comment:
Nit: for readability, since this variable has semantic meaning:

```suggestion
type AppHeaderBaseItem<Type extends string> = {
```